### PR TITLE
Integer#% speed up

### PR DESCRIPTION
**Pull request by miura1729**
Using bit operation.

### CODE REVIEW

**Comment by miura1729** [comment]
Sorry . This patch can't execute on non 32bit CPU. Please do not merge it.

**Comment by tsahara** [comment]
`#if` trick won't work on cross-compiled binaries in general.

**Comment by mattn** [comment]
You can replace the `#if`like below.
```c
#if UINTPTR_MAX == 0xffffffff // 32bit
#elif UINTPTR_MAX == 0xffffffffffffffff // 64bit
#else
// WTF
#endif
```

**Comment by miura1729** [comment]
I didn't think about cross-compile. I have no idea about cross-compile.
I think this patch needs checking specification of modulo for negative number. But most CPU doesn't need check in fact. Most CPUs have modulo instruction I assume.

**Comment by beoran** [comment]
The modulus operator % is guaranteed to work and exist by virtue of the C standard on all architectures, including embedded 8 or 16 bits architectures or long word 128 bits or 256 bits architectures (which this PR all ignores). Most optimizing compilers can and do replace the modulus by a bit operation if possible. See: http://stackoverflow.com/questions/22446425/do-c-c-compilers-such-as-gcc-generally-optimize-modulo-by-a-constant-power-of

In other words, this pull request is a bad idea. "Premature optimization is the root of all evil" in programming. We can and should use C's % and let the compiler optimize it for us.

**Comment by matz** [comment]
Close #2763
